google-datacatalog-tableau-connector/src/google/datacatalog_connectors/tableau/scrape/authenticator.py

```diff
@@ -25,24 +29,29 @@ def authenticate(cls,
                      api_version,
                      username,
                      password,
-                     site=None):
+                     site_content_url=None):
 
         url = f'{server_address}/api/{api_version}/auth/signin'
 
         headers = {
-            'Content-Type': 'application/json',
-            'Accept': 'application/json'
+            'Content-Type': constants.JSON_CONTENT_TYPE,
+            'Accept': constants.JSON_CONTENT_TYPE
         }
 
         body = {
             'credentials': {
                 'name': username,
                 'password': password,
                 'site': {
-                    'contentUrl': site
+                    'contentUrl': site_content_url
                 }
             }
         }
 
-        return requests.post(url=url, headers=headers,
-                             json=body).json()['credentials']
+        response = requests.post(url=url, headers=headers, json=body).json()
+
+        if not response.get('credentials'):
+            logging.critical(response)
+            sys.exit(1)
```

Comment:
should you exit in the `authenticator` class?

This seems to be too interruptive... and probably should be done by the synchronizer class?
If there is a way to do a fallback authentication or retry, ideally this should return an exception.

WDYT?

Reply:
I thought about it before adding this statement... The problem is: if the authentication fails, nothing else can be done by the connector because there will be no metadata to process. It's even different from successfully reading empty metadata, which means Data Catalog should at least be cleaned up. I don't see any fallback option. A retry mechanism, or better exception handling, could be implemented in a follow-up change.

In the scope of this PR, we can let it as is or revert the change so the connector will fail in subsequent statements (as used to happen before adding this), but in terms of the execution flow, we would get basically the same results. The benefit of adding this is that it lets the reason for the execution failure very clear for the user.

WDYT?

Reply:
I still think this is too interruptive... doing it in an inner layer should be avoided. Let's say this connector is used as a client library, we have this use case with the RDBMS and Hive connectors. Where another Python script calls the connector.

We are not sure how consumers may be using it, this `system.exit` may kill the whole Python process. Thinking better, even at the synchronizer code it might not be ideal. Maybe in the CLI code it would work... but raising the exception should be preferred.

I'd recommend removing it for now and improving this error handling scenario in a follow-up change.

Reply:
Agree, that makes sense. Reverted!